managed/resource_cluster.go

```diff
@@ -1018,16 +1024,17 @@ func (r resourceCluster) Create(ctx context.Context, req tfsdk.CreateResourceReq
 		return
 	}
 
-	credentials := openapiclient.NewCreateClusterRequestDbCredentialsWithDefaults()
+	createClusterRequest := *openapiclient.NewCreateClusterRequest(*clusterSpec)
+
+	encryptedCredentials := openapiclient.NewCreateClusterRequestEncryptedDbCredentialsWithDefaults()
 	if plan.Credentials.Username.IsNull() {
-		credentials.SetYsql(*openapiclient.NewDBCredentials(plan.Credentials.YSQLUsername.Value, plan.Credentials.YSQLPassword.Value))
-		credentials.SetYcql(*openapiclient.NewDBCredentials(plan.Credentials.YCQLUsername.Value, plan.Credentials.YCQLPassword.Value))
+		encryptedCredentials.SetYsql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.YSQLUsername.Value), b64(plan.Credentials.YSQLPassword.Value)))
+		encryptedCredentials.SetYcql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.YCQLUsername.Value), b64(plan.Credentials.YCQLPassword.Value)))
 	} else {
-		credentials.SetYsql(*openapiclient.NewDBCredentials(plan.Credentials.Username.Value, plan.Credentials.Password.Value))
-		credentials.SetYcql(*openapiclient.NewDBCredentials(plan.Credentials.Username.Value, plan.Credentials.Password.Value))
+		encryptedCredentials.SetYsql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.Username.Value), b64(plan.Credentials.Password.Value)))
+		encryptedCredentials.SetYcql(*openapiclient.NewEncryptedDBCredentials(b64(plan.Credentials.Username.Value), b64(plan.Credentials.Password.Value)))
 	}
-
-	createClusterRequest := *openapiclient.NewCreateClusterRequest(*clusterSpec, *credentials)
+	createClusterRequest.SetEncryptedDbCredentials(*encryptedCredentials)
```

Comment:
not sure but we might have an issue with this when we mark encrypted_db_credentials as required? Let's keep it as `createClusterRequest := *openapiclient.NewCreateClusterRequest(*clusterSpec, *encryptedCredentials)` only

Reply:
You can only pass required params there, optionals have to be added separately. 
```
managed/resource_cluster.go:1036:78: too many arguments in call to openapiclient.NewCreateClusterRequest
	have (openapi.ClusterSpec, openapi.CreateClusterRequestEncryptedDbCredentials)
	want (openapi.ClusterSpec)
make: *** [build] Error 1```

Reply:
Cool then will have to make the change later